src/Http/Requests/Messages/CreateMessageRequest.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace RonasIT\Chat\Http\Requests\Messages;
+
+use RonasIT\Chat\Contracts\Services\UserServiceContract;
+use RonasIT\Support\BaseRequest;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use RonasIT\Chat\Contracts\Requests\CreateMessageRequestContract;
+
+class CreateMessageRequest extends BaseRequest implements CreateMessageRequestContract
+{
+    protected $targetUser;
+
+    public function rules(): array
+    {
+        $mediaTableName = config('chat.database.tables.media');
+
+        return [
+            'recipient_id' => 'required|integer|filled',
+            'text' => 'string|required',
+            'attachment_id' => "integer|exists:{$mediaTableName},id",
+        ];
+    }
+
+    public function validateResolved()
+    {
+        $this->init();
+
+        parent::validateResolved();
+
+        $this->checkTargetUserExists();
+
+        $this->checkSelfMessage();
+    }
+
+    protected function init()
+    {
+        $this->targetUser = app(UserServiceContract::class)->find($this->input('recipient_id'));
+    }
+
+    protected function checkTargetUserExists()
+    {
+        if (empty($this->targetUser)) {
```

Comment:
we can validate `recipient_id` field via the `exists` validation rule without the user service and user service contract